### ISSUE TITLE
memory leak while unmounting the lottie component 

### ISSUE THREAD

**Issue by pSapien** [lottie-react-native/lottie-react-native#1010]
### Description

The Lottie React Native component seems to be causing a memory leak when unmounted. After the component is unmounted, the device does not release the memory, which causes the RAM usage to spike up.
Haven't checked in Android devices yet. 

### Steps to Reproduce

1. Mount a list of Lottie components inside a modal-like component
2. Unmount the component continuously
3. Watch the RAM level spike up

- Code

```javascript
import {useState, useRef, useEffect} from 'react';
import {View, Text, StatusBar} from 'react-native';
import LottieView from 'lottie-react-native';

StatusBar.setHidden(true);

const lottieImage = require('./assets/money_mouth_face.json');
const RE_RENDER_LIMIT = 50;
const INTERVAL_LIMIT = 1500;
const LOTTIE_VIEW_COUNT = 12;

export default function App() {
  const [show, setShow] = useState(false);
  const renderCountRef = useRef(0);

  useEffect(() => {
    const lastIntervalRef = setInterval(() => {
      if (renderCountRef.current >= RE_RENDER_LIMIT) {
        setShow(false);
        clearInterval(lastIntervalRef);
      } else {
        renderCountRef.current++;
        setShow(p => !p);
      }
    }, INTERVAL_LIMIT);
  }, []);

  if (renderCountRef.current >= RE_RENDER_LIMIT) return <Text>Done...</Text>;

  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      {show ? (
        Array.from({length: LOTTIE_VIEW_COUNT}).map((p, idx) => {
          return (
            <LottieView
              source={lottieImage}
              style={{height: 40, width: 40}}
              key={idx}
              autoPlay
              renderMode="HARDWARE"
            />
          );
        })
      ) : (
        <Text>Re-rendering...</Text>
      )}
    </View>
  );
}
```

Repo: https://github.com/pSapien/lottie-memory-leak

https://user-images.githubusercontent.com/26534891/232324421-0737fd99-9731-482d-b6b5-7be5c644b282.mov

**Expected behavior:** 

- When the Lottie React Native component is unmounted, I expect the device to release the memory used by the component, which should prevent the app's RAM usage from spiking up.

**Actual behavior:** 

When the Lottie React Native component is unmounted, the device does not release the memory used by the component, causing the memory leak.

### Versions

lottie-react-native: 6.0.0-rc.3, 6.0.0-rc.2, 5.1.5

**Comment by nikitph** [comment]
This is a very accurate representation of things. We have been using animationref.current. Reset() on unmount but this has to be automatic process.

**Comment by pSapien** [comment]
> This is a very accurate representation of things. We have been using animationref.current. Reset() on unmount but this has to be an automatic process.

Well, I went ahead and wrote this in the `LottieView` library component itself

```
  componentWillUnmount(): void {
    if (this.props.autoPlay) this.reset();
  }
```

Although the `reset` command is being triggered, the problem still persists.

**Comment by nikitph** [comment]
ahh i m not sure of the behavior with class components. we use it like this
in conjugation with react-navigation. This is how it looks like.

**Comment by nikitph** [comment]
useFocusEffect(
  useCallback(() => {
    if (animation.current) {
      animation.current.play();
    }
    return () => {
      animation.current && animation.current.reset();
    };
  }, []),
);

**Comment by matinzd** [comment]
Thank you for the detailed explanation. I will try to look into that on the weekend.

**Comment by pSapien** [comment]
> Thank you for the detailed explanation. I will try to look into that on the weekend.
> 
> However, the `reset` method does not unmount the view. It just resets the frame. @nikitph
> 
> Do you have new architecture turned on? Meanwhile, can you try adding `removeClippedSubviews` on the outer view and see how it behaves? @pSapien

Thanks for your quick reply.

In our organizational code, the new architecture is not turned on.  I could see no changes by adding `removeClippedSubviews` though

**Comment by bolan9999** [comment]
Any solution for this?

**Comment by Nsquik** [comment]
Any news @matinzd ?

**Comment by matinzd** [comment]
> Any news @matinzd ?

I put some time into that but still not luck. I will try again on the weekend.

**Comment by nikitph** [comment]
@matinzd would really appreciate an update on this. The library is working near perfect for us except for this problem.

**Comment by matinzd** [comment]
Now, we have addressed the issue and the pull request is in review.

Waiting for @emilioicai with higher access to approve.

**Comment by matinzd** [comment]
The fix is included in v6.0.0-rc.8:

https://github.com/lottie-react-native/lottie-react-native/releases/tag/v6.0.0-rc.8
